api/src/accounts.rs

```diff
@@ -251,6 +296,83 @@ impl Account {
+                    &StateKey::resource(&self.address.into(), &coin_store_type_tag).map_err(|err| {
+                        BasicErrorWith404::internal_with_code(
+                            err,
+                            AptosErrorCode::InternalError,
+                            &self.latest_ledger_info,
+                        )
+                    })?,
+                    self.ledger_version,
+                    &self.latest_ledger_info,
+                )?;
+                let coin_balance = match state_value {
+                    None => 0,
+                    Some(bytes) => bcs::from_bytes::<CoinStoreResourceUntyped>(&bytes).map_err(|err| {
+                        BasicErrorWith404::internal_with_code(
+                            err,
+                            AptosErrorCode::InternalError,
+                            &self.latest_ledger_info,
+                        )
+                    })?.coin()
+                };
+                (get_paired_fa_metadata_address(&move_struct_tag), coin_balance)
+            },
+            AssetType::FungibleAsset(fa_metadata_adddress) => (fa_metadata_adddress.into(), 0)
+        };
+        let primary_fungible_store_address = get_paired_fa_primary_store_address(self.address.into(), fa_metadata_address);
+        if let Some(data_blob) = self.context.get_state_value_poem(&StateKey::resource_group(&primary_fungible_store_address, &ObjectGroupResource::struct_tag()), self.ledger_version, &self.latest_ledger_info)? {
+            if let Some(object_group) = bcs::from_bytes::<ObjectGroupResource>(&data_blob).ok() {
+                if let Some(fa_store) = object_group.group.get(&FungibleStoreResource::struct_tag()) {
+                    let fa_store_resource = bcs::from_bytes::<FungibleStoreResource>(fa_store).map_err(|err| {
+                        BasicErrorWith404::internal_with_code(
+                            err,
+                            AptosErrorCode::InternalError,
+                            &self.latest_ledger_info,
+                        )
+                    })?;
+                    if fa_store_resource.balance != 0 {
+                        balance += fa_store_resource.balance();
+                    } else if let Some(concurrent_fa_balance) = object_group.group.get(&ConcurrentFungibleBalanceResource::struct_tag()) {
+                        // query potential concurrent fa balance
+                        let concurrent_fa_balance_resource = bcs::from_bytes::<ConcurrentFungibleBalanceResource>(concurrent_fa_balance).map_err(|err| {
+                            BasicErrorWith404::internal_with_code(
+                                err,
+                                AptosErrorCode::InternalError,
+                                &self.latest_ledger_info,
+                            )
+                        })?;
+                        balance += concurrent_fa_balance_resource.balance();
+                    }
```

Comment:
How's this compare in performance to the view function?

Also, if we're doing this... we probably want a supply query too

Reply:
good question. it does not need move vm so I guess it should be much better.
Anyway to quickly test it?

Reply:
I was just wondering if you had a comparison.

Otherwise, this should be much cheaper.  I don't need you to prove it if you didn't compare.